src/lib/y2storage/autoinst_proposal.rb

```diff
@@ -101,24 +101,48 @@ def propose_devicegraph(devicegraph, drives)
 
     # Add partition tables
     #
-    # This method create partitions table for disks which are missing one.
+    # This method create/change partitions tables according to information
+    # specified in the profile. Disks containing any partition will be ignored.
+    #
     # The devicegraph which is passed as first argument will be modified.
     #
     # @param devicegraph [Devicegraph]                 Starting point
     # @param drives      [Proposal::AutoinstDrivesMap] Devices map from an AutoYaST profile
     def add_partition_tables(devicegraph, drives)
       drives.each do |disk_name, drive_spec|
         disk = devicegraph.disk_devices.find { |d| d.name == disk_name }
-        next if disk.nil? || disk.partition_table
-
-        ptable_type =
-          if drive_spec.disklabel
-            Y2Storage::PartitionTables::Type.find(drive_spec.disklabel)
-          else
-            Y2Storage::PartitionTables::Type::MSDOS
-          end
-        disk.create_partition_table(ptable_type)
+        next if disk.nil? || !disk.partitions.empty?
+
+        update_partition_table(disk, suitable_ptable_type(disk, drive_spec))
+      end
+    end
+
+    # Determines which partition table type should be used
+    #
+    # @param disk        [Y2Storage::Disk] Disk to set the partition table on
+    # @param drive_spec  [Y2Storage::AutoinstProfile::DriveSection] Drive section from the profile
+    # @return [Y2Storage::PartitionTables::Type] Partition table type
+    def suitable_ptable_type(disk, drive_spec)
+      ptable_type = nil
+      if drive_spec.disklabel
+        ptable_type = Y2Storage::PartitionTables::Type.find(drive_spec.disklabel)
       end
+
+      disk_ptable_type = disk.partition_table ? disk.partition_table.type : nil
+      ptable_type || disk_ptable_type || Y2Storage::PartitionTables::Type::MSDOS
```

Comment:
Only for curiosity, what happens when we are using UEFI or DASD? Does it fallback to msdos? In that case, probably it fails when tries to create the partition table.

Reply:
In fact, we have a method `Partitionable#preferred_ptable_type`

https://github.com/yast/yast-storage-ng/blob/master/src/lib/y2storage/partitionable.rb#L246

Reply:
Oh, I didn't new about that method. Maybe AutoYaST should rely on it. Thanks!